pkgs/by-name/fo/foxglove-studio/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  alsa-lib,
+  at-spi2-atk,
+  autoPatchelfHook,
+  cairo,
+  cups,
+  dbus,
+  dpkg,
+  eudev,
+  expat,
+  fetchurl,
+  glib,
+  glibc,
+  gtk3,
+  lib,
+  libdrm,
+  libgcc,
+  libxkbcommon,
+  makeWrapper,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  stdenv,
+  systemd,
+  via,
+  xorg,
+}:
+let
+  libraries = [
+    alsa-lib
+    at-spi2-atk
+    cairo
+    cups
+    dbus
+    eudev
+    expat
+    glib
+    glibc
+    gtk3
+    libdrm
+    libgcc.libgcc
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    systemd
+    via
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXrandr
+    xorg.libxcb
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  name = "foxglove-studio";
+  version = "2.18.1";
+
+  src = fetchurl {
+    url = "https://get.foxglove.dev/desktop/v${finalAttrs.version}/foxglove-studio-latest-linux-amd64.deb";
+    sha256 = "sha256-vHloPOmrr8HR9Su5s84xohh8uCpbG4D9eEqbp+jPsDc=";
+  };
+  
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
+  ];
+
+  buildInputs = libraries;
+
+  preBuild = ''
+    addAutoPatchelfSearchPath $out/opt/Foxglove Studio/
+  '';
+
+  unpackPhase = ''
+    dpkg-deb -x $src .
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -R usr/share opt $out/
+
+    # Entrypoint
+    mkdir -p $out/bin
+    makeWrapper "$out/opt/Foxglove Studio/foxglove-studio" $out/bin/foxglove-studio \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath libraries}"
+
+
+    # fix the path in the desktop file
+    substituteInPlace \
+      $out/share/applications/foxglove-studio.desktop \
+      --replace /opt/ $out/opt/
+
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://foxglove.dev/";
+    description = "Platform enabling robotics teams to record, process, visualize, and collaborate on multimodal robot data";
+    license = lib.licenses.unfree;
+    downloadPage = "https://foxglove.dev/download";
+    changelog = "https://docs.foxglove.dev/changelog/foxglove/v${finalAttrs.version}";
+    platforms = lib.platforms.all;
+    maintainers = with lib.maintainers; [ lennartrth ];
```

Comment:
Should also add `meta.sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];` as it's repackaged third-party native code